srtcore/channel.cpp

```diff
@@ -386,20 +400,26 @@ int CChannel::sockoptQuery(int SRT_ATR_UNUSED level, int SRT_ATR_UNUSED option)
     return -1;
 }
 
-void CChannel::getSockAddr(sockaddr* addr) const
+void CChannel::getSockAddr(sockaddr_any& w_addr) const
 {
-   socklen_t namelen = m_iSockAddrSize;
-   ::getsockname(m_iSocket, addr, &namelen);
+    // The getsockname function requires only to have enough target
+    // space to copy the socket name, it doesn't have to be corelated
+    // with the address family. So the maximum space for any name,
+    // regardless of the family, does the job.
+    socklen_t namelen = sizeof(w_addr);
```

Comment:
Well, this is some feature coming from UDT and I doubt anyone is using it. The use method is specific, though. You should have some UDP socket already created and configured some specific way, but then if you use it with `srt_bind_peerof` (there's upcoming PR that renames it to a more suitable name `srt_bind_acquire`) and generally forget it (CChannel class takes ownership over it, any variable holding it should get overwritten and this socket should never be `::close`d). Setting these option as before is required due to specifics of SRT, as it is required anyway and the code can't rely on that the user has configured the socket properly and possibly made SRT go out of control.

Generally this feature's use is that "create a UDP socket and configure it the best way you think is right - then give it to SRT and we'll do the rest, you shouldn't touch it anymore".

Reply:
If I use `sizeof(sockaddr)` then it will definitely cease to work.

It is a known feature of `sockaddr_any` that it wraps all sockaddr* structures in a union and therefore the total size is enough to handle any of the families. In this particular case it may also embrace the extra `len` field, but it doesn't count - what counts is that this size is big enough to copy back the contents of `sockaddr_in` or `sockaddr_in6` structure.

The closest approximation of what you might need would be:
```
socklen_t namelen = max<socklen_t>(sizeof sockaddr_in, sizeof sockaddr_in6);
```
An alternative solution is to make a static function in `sockaddr_any` that can return the size of the sockaddr union.

As it comes for the expression of `&w_addr` used in the `getsockname` call, note that it uses the overloaded `operator&` function here. The intention is obvious enough because in C++ this wouldn't compile otherwise.